work_queue/src/work_queue.c

```diff
@@ -3965,6 +3968,59 @@ static struct work_queue_worker *find_worker_by_time(struct work_queue *q, struc
 	}
 }
 
+
+// checks workers against tasks and returns if a task has the  
+// possibility of fitting a worker
+static int check_task_fit_in_worker(struct work_queue *q, struct work_queue_task *t, struct work_queue_worker *w)
+{
+	int set = 0;
+	struct rmsummary *l = task_worker_box_size(q,w,t);
+
+	// baseline resurce comparison of worker total resources and a task requested resorces
+
+	if(w->resources->cores.total < l->cores ) {
+		set = set | (1<<0);
+	}
+
+	if(w->resources->memory.total < l->memory ) {
+		set = set | (1<<1);
+	}
+
+	if(w->resources->disk.total < l->disk ) { 
+		set = set | (1<<2);
+	}
+
+	if(w->resources->gpus.total < l->gpus ) {
+		set = set | (1<<3);
+	}
+	rmsummary_delete(l);
+	return set;
+}
+static int check_task_fit_in_connected_workers(struct work_queue *q, struct work_queue_task *t)
+{
+	char *key;
+	struct work_queue_worker *w;
+	hash_table_firstkey(q->worker_table);
+	
+	int bit_set = 0;
+	while(hash_table_nextkey(q->worker_table, &key, (void**)&w))
+	{
+		int new_set = check_task_fit_in_worker(q, t, w);
+		
+		if (bit_set == 0){
+			// Task could run on a currently connected worker, immediately return	
+			return 1;
+		}
+
+		// Inherit the unfit criteria for this task
+		bit_set = bit_set | new_set;
```

Comment:
Clever... but it may give confusing information to the user.   Say

my task is: 2 cores, 10 GB of memory
worker A: 1 core, 1 GB of memory
worker B: 1 core, 20 GB of memory

The message for memory will read:
1 waiting tasks need more than 20000 MB of memory(max available). The largest task needs 10000 MB of memroy